q2_feature_classifier/_skl.py

```diff
@@ -75,13 +74,13 @@ def _predict_chunk_with_conf(pipeline, separator, confidence, chunk):
     split_classes = [c.split(separator) for c in pipeline.classes_]
     for seq_id, taxon, class_probs in zip(seq_ids, y, prob_pos):
         taxon = taxon.split(separator)
-        classes = zip(deepcopy(split_classes), class_probs)
+        classes = zip(split_classes, class_probs)
         result = []
-        for level in taxon:
-            classes = [cls for cls in classes if cls[0].pop(0) == level]
-            cum_prob = sum(c[1] for c in classes)
+        for rank, level in enumerate(taxon):
+            cum_prob = sum(cls[1] for cls in classes if cls[0][rank] == level)
```

Comment:
This is the potential error. The error arises because sometimes taxa match at a later lever (in the label) but differ earlier. For instance the Ruminococcus genus exists in both the Lachnospiraceae and Ruminococcaceae families in Greengenes.

I suggest matching on all the levels up to the current one. Something like
```
for rank in range(1, len(taxon) + 1):
    cum_prob = sum(cls[1] for cls in classes if cls[0][:rank] == taxon[:rank])
```